vmm/src/interrupt.rs

```diff
@@ -177,18 +177,29 @@ impl InterruptSourceGroup for MsiInterruptGroup {
                 route: self.vm.make_routing_entry(route.gsi, &config),
                 masked,
             };
+
+            // When mask a msi irq, entry.masked is set to be true,
+            // and the gsi will not be passed to KVM through KVM_SET_GSI_ROUTING.
+            // so it's safe to call disable() which deassign KVM_IRQFD
+            // before set_gsi_routes().
```

Comment:
From your comment earlier, it believe it should be:

`So it's required to call disable() (which deassign KVM_IRQFD) before set_gsi_routes() to avoid kernel panic (see #3827)`

Reply:
Many thanks.
PR updated as you suggested.